### PR TITLE
SNOW-698806: add usage telemetry information

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-69880

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   set connection application so that snowflake knows the source

### CODE REVIEW

File: src/snowflake/sqlalchemy/util.py [hunk 1]
```diff
@@ -85,3 +94,15 @@ def _set_connection_interpolate_empty_sequences(
     else:
         # _dbapi_connection is a raw SnowflakeConnection
         dbapi_connection._interpolate_empty_sequences = flag
+
+
+def _update_connection_application_name(
+    **conn_kwargs: Dict[str, Any]
```
Comment (sfc-gh-mkeller) [inline src/snowflake/sqlalchemy/util.py:100]: I think `Any` is enough here according to https://peps.python.org/pep-0484/#arbitrary-argument-lists-and-default-argument-values
```suggestion
    **conn_kwargs: Any
```

File: DESCRIPTION.md [hunk 2]
```diff
@@ -9,6 +9,10 @@ Source code is also available at:
 
 # Release Notes
 
+- v1.4.5(Dec 7, 2022)
+
+  - Added telemetry for sqlalchemy usage information.
```
Comment (sfc-gh-mkeller) [inline DESCRIPTION.md:14]: Could please rewrite this? I know the application name is used for telemetry too, but really you updated the application name given to the connector

**Review by sfc-gh-mkeller — APPROVED** [review]
:ship: :ship:

File: src/snowflake/sqlalchemy/util.py (hunk 1 above)
Comment (sfc-gh-aling) [reply src/snowflake/sqlalchemy/util.py]: updated

File: DESCRIPTION.md (hunk 2 above)
Comment (sfc-gh-aling) [reply DESCRIPTION.md]: you're correct, I've rewritten this.

**Pull request merged**